cpp/src/arrow/compute/light_array.h

```diff
@@ -31,6 +33,18 @@
 namespace arrow {
 namespace compute {
 
+/// \brief Context needed by various execution engine operations
+///
+/// In the execution engine this context is provided by either the node or the
+/// plan and the context exists for the lifetime of the plan.  Defining this here
+/// allows us to take advantage of these resources without coupling the logic with
+/// the execution engine.
+struct LightContext {
+  bool has_avx2() const { return (hardware_flags & arrow::internal::CpuInfo::AVX2) > 0; }
```

Comment:
Why is this no using `CpuInfo::IsSupported(CpuInfo::AVX2)`?

Reply:
IIRC, the concept here was to be able to attach hardware flags to a specific context rather than needing to disable or enable for the whole library using `CpuInfo::EnableFeature()`. It and many other things are certainly candidates for follow up refactoring